src/core/common/bit_vector.hpp

```diff
@@ -78,15 +78,24 @@ template <uint16_t N> class BitVector : public Equatable<BitVector<N>>
     }
 
     /**
-     * This method sets the mask of a given index.
+     * This method sets or clears the mask of a given index.
      *
      * @param[in] aIndex  The index.
+     * @param[in] aSet    TRUE to sets the mask, or FALSE to clear the mask.
      *
      */
-    void Set(uint16_t aIndex)
+    void Set(uint16_t aIndex, bool aSet = true)
```

Comment:
There are two models here: (1) `Set()` with a param for the value and (2) `Set()`/`Clear()`. Option 1 here would seem to add code for those cases where the caller does not conditionally call `Set()` vs. `Clear()`. However, it may not make any difference if the caller always has its own if-statement. Thoughts?

Reply:
It makes the caller code simpler in some cases, as in this PR. 

If the caller has its own if-statement, it can call `Set()` and `Clear()` for different conditions. 

I agree it is a little bit confusing to have `Set(false)` and `Clear()` do the same job. 
Maybe we should remove `Clear()`  and always use `Set(bool)`. If so, we can let BitVector inherit Clearable and use `Clear()` instead of `ClearAll()`. 
Thoughts? @jwhui @abtink @librasungirl

Reply:
I think it's too much change for `BitVector` in this PR. 
I will revert the BitVector to have separate `Set()` and `Clear()` in this PR. 

But let's continue discussing on this subject here to find out the best solution. 
Thoughts?

Reply:
Moving the `BitVector()` change to a separate PR sounds good. It will allow us to better evaluate the pros/cons of the change.

My only concern with having a single `Set()` method is that this copies the code everywhere (even in cases where the conditional is not necessary). Is this true given that the methods are implemented in a header? Or will the compiler properly optimize this?

Reply:
I think it's fine if `Set(true)` and `Set(false)` is as efficient as current `Set()` and `Clear()` (optimized by compiler).
We can see from `size-report` if this is too in the BitVector refactoring PR.